### DISCUSSION TITLE
AllureWriter in cypress 10

### DISCUSSION THREAD

**Discussion by pwpatton** [Shelex/cypress-allure-plugin#152] [Q&A] [answered]
Hello @Shelex, I'm unclear on how to get `AllureWriter` and cypress 10 to work. 

In your docs you say to do it like this for cypress 10:
```
module.exports = defineConfig({
    e2e: {
        setupNodeEvents(on, config) {
            AllureWriter(on, config);
            return config;
        }
    }
});
```

But my IDE cannot find AllureWriter and it's not found when running cypress either.

```
ReferenceError: AllureWriter is not defined
    at setupNodeEvents (c:\Landmark\sandbox1\web\client\LawsonWebClient\WebContent\cli\cypress\cypress.config.ts:28:7)
```

TIA for any help you can give.

**Comment by Shelex** [comment]
Hi @pwpatton !
Thank you for your question.

Actually without a context there may be multiple things that may go wrong.
First that comes to my mind - if AllureWriter "is not defined", it should be defined then by `import allureWriter from '@shelex/cypress-allure-plugin/writer';`.
If you already have imported a writer - maybe you can provide an example where such issue could be reproduced?

**Comment by pwpatton** [comment, marked as answer]
Ok, looks like user error on my part. I was confused why in the docs A in `AllureWriter` was capitalized but what I really needed to use was `allureWriter`.

My cypress.config.ts now looks like this and works great. Thanks @Shelex 

```typescript
import { defineConfig } from 'cypress';
const allureWriter = require('@shelex/cypress-allure-plugin/writer');

export default defineConfig({
  env: {
    allure: true,
    allureLogCypress: true,
    allureClearSkippedTests: true
  },
  e2e: {
    setupNodeEvents: (on: Cypress.PluginEvents, config: Cypress.PluginConfigOptions) => {
      allureWriter(on, config);
      return config;
    }
  }
});
```

**Comment by thiagocosenza** [comment]
Hello @Shelex and @pwpatton , I tested with the 'a' not capitalized as well and still get the error.

My cypress.config.js:

```
const { defineConfig } = require('cypress');

module.exports = defineConfig({
  e2e: {
    setupNodeEvents(on, config) {
      AllureWriter(on, config);
      return config;
    },
    env: {
      allure: "true",
      allureResultsPath: "cypress/support/reports",
      allureAddAnalyticLabels: "true"
    }
});
```
and added **import '@shelex/cypress-allure-plugin';** to my e2e.js file.

It's returning: **ReferenceError: AllureWriter is not defined**

**Reply by Shelex** [reply]
> Hello @Shelex and @pwpatton , I tested with the 'a' not capitalized as well and still get the error.
> 
> My cypress.config.js:
> 
> ```
> const { defineConfig } = require('cypress');
> 
> module.exports = defineConfig({
>   e2e: {
>     setupNodeEvents(on, config) {
>       AllureWriter(on, config);
>       return config;
>     },
>     env: {
>       allure: "true",
>       allureResultsPath: "cypress/support/reports",
>       allureAddAnalyticLabels: "true"
>     }
> });
> ```
> 
> and added **import '@shelex/cypress-allure-plugin';** to my e2e.js file.
> 
> It's returning: **ReferenceError: AllureWriter is not defined**

do you have import statement in `cypress.config.js`
`import AllureWriter from '@shelex/cypress-allure-plugin/writer'`?

**Reply by thiagocosenza** [reply]
I'm not sure how to do this, as when I try:

```
const { defineConfig } = require('cypress');
**import AllureWriter from '@shelex/cypress-allure-plugin/writer'**

module.exports = defineConfig({
  e2e: {
    setupNodeEvents(on, config) {
      AllureWriter(on, config);
      return config;
    },
    env: {
      allure: "true",
      allureResultsPath: "cypress/support/reports",
      allureAddAnalyticLabels: "true",
    },
  },
});
```
it returns: 

```
import AllureWriter from '@shelex/cypress-allure-plugin/writer'
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at Object.compileFunction (node:vm:360:18)
    at wrapSafe (node:internal/modules/cjs/loader:1048:15)
    at Module._compile (node:internal/modules/cjs/loader:1083:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1173:10)
    at Module.load (node:internal/modules/cjs/loader:997:32)
    at Module._load (node:internal/modules/cjs/loader:838:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:170:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:527:24)
    at async importModuleDynamicallyWrapper (node:internal/vm/module:438:15)
    at async loadFile (/Users/tcosenza/Library/Caches/Cypress/10.8.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/child/run_require_async_child.js:106:14)
    at async EventEmitter.<anonymous> (/Users/tcosenza/Library/Caches/Cypress/10.8.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/child/run_require_async_child.js:116:32)
```

**Reply by Shelex** [reply]
> I'm not sure how to do this, as when I try:
> 
> ```
> const { defineConfig } = require('cypress');
> **import AllureWriter from '@shelex/cypress-allure-plugin/writer'**
> 
> module.exports = defineConfig({
>   e2e: {
>     setupNodeEvents(on, config) {
>       AllureWriter(on, config);
>       return config;
>     },
>     env: {
>       allure: "true",
>       allureResultsPath: "cypress/support/reports",
>       allureAddAnalyticLabels: "true",
>     },
>   },
> });
> ```
> 
> it returns:
> 
> ```
> import AllureWriter from '@shelex/cypress-allure-plugin/writer'
> ^^^^^^
> 
> SyntaxError: Cannot use import statement outside a module
>     at Object.compileFunction (node:vm:360:18)
>     at wrapSafe (node:internal/modules/cjs/loader:1048:15)
>     at Module._compile (node:internal/modules/cjs/loader:1083:27)
>     at Module._extensions..js (node:internal/modules/cjs/loader:1173:10)
>     at Module.load (node:internal/modules/cjs/loader:997:32)
>     at Module._load (node:internal/modules/cjs/loader:838:12)
>     at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:170:29)
>     at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
>     at async Promise.all (index 0)
>     at async ESMLoader.import (node:internal/modules/esm/loader:527:24)
>     at async importModuleDynamicallyWrapper (node:internal/vm/module:438:15)
>     at async loadFile (/Users/tcosenza/Library/Caches/Cypress/10.8.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/child/run_require_async_child.js:106:14)
>     at async EventEmitter.<anonymous> (/Users/tcosenza/Library/Caches/Cypress/10.8.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/child/run_require_async_child.js:116:32)
> ```

Oh, try `require` syntax then:
`const AllureWriter = require('@shelex/cypress-allure-plugin/writer');`

**Reply by thiagocosenza** [reply]
Oh, that's right.  Sorry for the simple mistake. It's all working good now! Thank you so much.
